src/components/slider.js

```diff
@@ -0,0 +1,371 @@
+      />
+    </svg>
+  );
+}
+
+class Slider extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    const { slideIndex = 0, direction = HORIZONTAL } = this.props;
+    this.state = {
+      currentSlideIndex: slideIndex,
+      animating: false
+    };
+    this.slideCount = React.Children.count(this.props.children);
+    this.direction = direction;
+    this.swipeProperty = direction === HORIZONTAL ? "left" : "top";
+    this.swipeEventProperty = direction === HORIZONTAL ? "clientX" : "clientY";
+  }
+
+  componentDidMount() {
+    this.setupAutoplay();
+  }
+
+  componentWillUnmount() {
+    this.stopAutoplay();
+    if (this.animationTimerId) {
+      clearTimeout(this.animationTimerId);
+    }
+  }
+
+  componentWillReceiveProps(props) {
+    this.slideCount = React.Children.count(props.children);
+    if (this.state.currentSlideIndex >= this.slideCount) {
+      this.setState({ currentSlideIndex: 0 });
+    }
+  }
+
+  setupAutoplay = () => {
+    if (this.props.autoplay && !this.isMouseOver) {
+      this.stopAutoplay();
+      this.autoplayTimerId = setInterval(
+        this.next,
+        parseInt(this.props.autoplay, 10)
+      );
+    }
+  };
+
+  stopAutoplay = () => {
+    if (this.autoplayTimerId) {
+      clearInterval(this.autoplayTimerId);
+    }
+  };
+
+  onAnimationEnd = () => {
+    this.setState({
+      currentSlideIndex: this.nextSlideIndex,
+      animating: false,
+      animation: undefined
+    });
+    this.setupAutoplay();
+    if (typeof this.props.onSlideChange === "function") {
+      this.props.onSlideChange({ slideIndex: this.nextSlideIndex });
+    }
+  };
+
+  isDisabled = () =>
+    this.slideCount < 2 || this.state.animating || this.props.disabled;
+
+  isInfinite = () => this.slideCount > 2 && this.props.infinite !== false;
+
+  canGoPrevious = () => this.isInfinite() || this.state.currentSlideIndex > 0;
+
+  canGoNext = () =>
+    this.isInfinite() || this.state.currentSlideIndex < this.slideCount - 1;
+
+  goTo = (index, animation) => {
+    if (this.isDisabled()) return;
+    this.nextSlideIndex = index;
+    this.setState({ animating: true, animation });
+    const timeout = this.props.duration || DEFAULT_DURATION;
+    this.animationTimerId = setTimeout(this.onAnimationEnd, timeout);
+  };
+
+  previous = () => {
+    if (!this.canGoPrevious()) return;
+    const nextSlideIndex = this.state.currentSlideIndex - 1;
+    const actualNextSlide =
+      nextSlideIndex >= 0 ? nextSlideIndex : this.slideCount - 1;
+    this.goTo(actualNextSlide, PREVIOUS);
+  };
+
+  next = () => {
+    if (!this.canGoNext()) return;
+    const nextSlideIndex = (this.state.currentSlideIndex + 1) % this.slideCount;
+    this.goTo(nextSlideIndex, NEXT);
+  };
+
+  getSlideClass = index => {
+    const { currentSlideIndex, animation } = this.state;
+    const classNames = this.getClassNames();
+    const lastSlideIndex = this.slideCount - 1;
+    if (index === currentSlideIndex) {
```

Comment:
For readability, lets keep conditional statements on the same line and if possible, convert these multiple `if`'s into a switch.